lib/utils.sh

```diff
@@ -63,3 +57,27 @@ filter_version_candidates() {
   done
 }
 
+versions_cache_dir="${ASDF_DATA_DIR:-${ASDF_HOME:-$HOME/.asdf}}/tmp/$(plugin_name)/cache"
+mkdir -p $versions_cache_dir
+
+etag_file="$versions_cache_dir/etag"
+index_file="$versions_cache_dir/index"
+touch "$etag_file" "$index_file"
+
+print_index_tab(){
+  local temp_headers_file="$(mktemp)"
+
+  if [ -f "$etag_file" ]; then
+    etag_flag='--header If-None-Match:'"$(cat "$etag_file")"
+  fi
+
+  index="$(curl --fail --silent --dump-header "$temp_headers_file" $etag_flag  "${NODEJS_ORG_MIRROR}index.tab")"
+  if [ -z "$index" ]; then
+    cat "$index_file"
+  else
+    cat "$temp_headers_file" | awk '/^etag: /{print(substr($0, 7))}' > "$etag_file"
```

Comment:
I think is better to normalize the etag before comparing it, because its specification is case insensitive, so both ETag and etag are valid.

Because awk splits the line into fields by space we can just test $1 to lowercase etag instead of a regex, and print $2 instead substr

```suggestion
    cat "$temp_headers_file" | awk 'tolower($1) == "etag:" { print $2 }' > "$etag_file"
```